app/assets/javascripts/helpers/profile_photo_uploader.es6

```diff
@@ -0,0 +1,250 @@
+    this.cropContainer = document.querySelector(".crop-container");
+    this.spinner = document.querySelector("#file-upload-spinner");
+
+    /**
+     * Creates a button
+     * @param {string} icon - The entypo icon class.
+     * @param {function} onClick - Is called when button has been clicked.
+     */
+    this.createButton = (icon, onClick) =>
+      ($(`<button class="btn btn-default" type="button"><i class="entypo-${icon}"></i></button>`)
+        .on("click", onClick));
+
+    /**
+     * Shows a message using flash messages or alert for mobile.
+     * @param {string} type - The type of the message, e.g. "error" or "success".
+     * @param text - The text to display.
+     */
+    this.showMessage = (type, text) => { return app.flashMessages ? app.flashMessages[type](text) : alert(text); };
+
+    this.initFineUploader();
+  }
+
+  /**
+   * Initializes the fine uploader component
+   */
+  initFineUploader() {
+    this.fineUploader = new qq.FineUploaderBasic({
+      element: this.fileInput,
+      validation: {
+        allowedExtensions: ["jpg", "jpeg", "png"]
+      },
+      request: {
+        endpoint: Routes.photos(),
+        params: {
+          /* eslint-disable camelcase */
+          authenticity_token: $("meta[name='csrf-token']").attr("content"),
+          /* eslint-enable camelcase */
+          photo: {"pending": true, "aspect_ids": "all", "set_profile_photo": true}
+        }
+      },
+      button: this.fileInput,
+      autoUpload: false,
+
+      messages: {
+        typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
+        sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
+        emptyError: Diaspora.I18n.t("photo_uploader.empty")
+      },
+
+      callbacks: {
+        onProgress:
+          (id, fileName, loaded, total) => (this.info.innerText = `${fileName} ${Math.round(loaded / total * 100)}%`),
+        onSubmit: (id, name) => this.onPictureSelected(id, name),
+        onComplete: (id, fileName, responseJSON) => this.onUploadCompleted(id, fileName, responseJSON),
+        onError: (id, name, errorReason) => this.showMessage("error", errorReason)
+      },
+
+      text: {
+        fileInputTitle: ""
+      },
+
+      scaling: {
+        sendOriginal: false,
+
+        sizes: [
+          {maxSize: 1600}
+        ]
+      }
+    });
+  }
+
+  /**
+   * Called when a picture from user's device has been selected.
+   * @param {number} id - The current file's id.
+   * @param {string} name - The current file's name.
+   */
+  onPictureSelected(id, name) {
+    this.setLoading(true);
+    this.fileName = name;
+    const file = this.fileInput.querySelector("input").files[0];
+
+    // ensure browser's file reader support
+    if (FileReader && file) {
+      try {
+        const fileReader = new FileReader();
+        fileReader.onload = () => this.initCropper(fileReader.result);
+        fileReader.readAsDataURL(file);
+      } catch (e) {
+        this.setLoading(false);
```

Comment:
Hm, when could `readAsDataURL` throw? Is this something that could occur during normal use? I cannot think of any scenario where it might.

So maybe, catching isn't the best idea here, since we'll lose any way of debugging stuff if uploading an image fails from the user perspective. Maybe `finally` would be more suitable to ensure the loading animation is disabled, but have the error be actually thrown.

Reply:
you're right, `readAsDataURL` cannot throw here and the existence of `FileReader` is checked, so no need for `try catch`. I just pushed an update and fixed this.